include/pgduckdb/logger.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+extern "C" {
+bool errstart(int elevel, const char *domain);
+void errfinish(const char *filename, int lineno, const char *funcname);
+int	errmsg_internal(const char *fmt,...);
+}
+
+namespace pgduckdb {
+
+/* PG Error level codes */
+#define DEBUG5		10
+#define DEBUG4		11
+#define DEBUG3		12
+#define DEBUG2		13
+#define DEBUG1		14
+#define LOG			15
+#define INFO		17
+#define NOTICE		18
+#define WARNING		19
+#define PGWARNING	19
+#define WARNING_CLIENT_ONLY	20
+
+// From PG elog.h
+#if defined(errno) && defined(__linux__)
+#define pd_prevent_errno_in_scope() int __errno_location pg_attribute_unused()
+#elif defined(errno) && (defined(__darwin__) || defined(__FreeBSD__))
+#define pd_prevent_errno_in_scope() int __error pg_attribute_unused()
+#else
+#define pd_prevent_errno_in_scope()
+#endif
+
+#define pd_ereport_domain(elevel, domain, ...)	\
+	do { \
+		pd_prevent_errno_in_scope(); \
+		static_assert(elevel >= DEBUG5 && elevel <= WARNING_CLIENT_ONLY, "Invalid error level"); \
+		if (errstart(elevel, domain)) \
+			__VA_ARGS__, errfinish(__FILE__, __LINE__, __func__); \
+	} while(0)
+
+
+#define pd_ereport(elevel, ...)	\
+	pd_ereport_domain(elevel, TEXTDOMAIN, __VA_ARGS__)
+
+#define pdlog(elevel, ...)  \
```

Comment:
`pd` stands for **p**g_**d**uckdb?

I don't think we use that acronym anywhere else. I do agree that we want something short, so maybe it's the right time to start using it.

let's stay consistent then with the `_` at least.

Reply:
Yeah exactly - I wanted something as concise as possible (also why I omitted the `_` since it will be so pervasive. But agree it's probably better to stay consistent.